CenterForOpenScience/ember-osf-web
# [ENG-1384][Feature] Add license metadata field

- Ticket: [ENG-1384]
- Feature flag: n/a

## Purpose

To add the license widget to the metadata form on the registries submission page

## Summary of Changes

- Add manager for the metadata registries license picker instance
- Update the license picker to handle a model or a changeset
- Update mirage draft-registration instances to also include a provider
- Update tests that use a draft-registration to also have a provider

<img width="615" alt="Screen Shot 2020-01-15 at 4 47 07 PM" src="https://user-images.githubusercontent.com/19379783/72474425-c52f2c00-37b6-11ea-945d-1f04dc54a516.png">

Note that the buttons were removed in this case from the license-picker

## Side Effects

This will add a license picker to the metadata section on the registries submission form, but also has some changes to the pre-existing license-picker and the editable license-picker manager. Changes here might also affect other areas that include the license-picker/editable license-picker manager.

## QA Notes

This should be tested as part of the metadata form fields. Does have changes to the pre-existing registries license-picker component and the editable license-picker manager. Both the component and the manager are only also used on the registries overview page, and should probably also be tested to ensure that everything still works as it should.


[ENG-1384]: https://openscience.atlassian.net/browse/ENG-1384
